### ISSUE TITLE
[a11y] Range slider track not visible in High Contrast

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#796]
## Issue summary

The [Range slider](https://polaris.shopify.com/components/forms/range-slider#navigation) component's track is not visible when the page is viewed using Windows High Contrast. This is because the track relies on  background color alone to convey its shape and size. Low vision users who rely on High Contrast may have difficulty using the control as a result.

![Range slider](https://user-images.githubusercontent.com/1462085/41673571-2beda9b2-7472-11e8-9b7b-8004b7e3b48a.png)

![Range slider in Edge using Windows High Contrast](https://user-images.githubusercontent.com/1462085/41673574-2dcac90e-7472-11e8-8c3d-830c6e1f5427.png)

#### Range slider style (CSS)

```
.Polaris-RangeSlider__Input {
  margin: 0;
  padding: 0;
  width: 100%;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 20;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}
```

#### Range slider track style (CSS)

```
.Polaris-RangeSlider__Input::-webkit-slider-runnable-track {
  --progress-lower: #5c6ac4;
  --progress-upper: #c4cdd5;
  cursor: pointer;
  width: 100%;
  height: .4rem;
  background-color: var(--progress-upper,#c4cdd5);
  background-image: linear-gradient(90deg,var(--progress-lower,transparent) 0,var(--progress-lower,transparent) var(--Polaris-RangeSlider-progress,0),var(--progress-upper,transparent) var(--Polaris-RangeSlider-progress,100%),var(--progress-upper,transparent));
  border: none;
  border-radius: .4rem;
}
```

## Steps to reproduce

1. Navigate to the [Range slider](https://polaris.shopify.com/components/forms/range-slider#navigation) component using Edge and do a visual inspection of the slider.
1. Turn on any a [High Contrast theme in Windows](https://support.microsoft.com/en-us/help/13862/windows-use-high-contrast-mode) and redo the visual inspection.

### Expected

- Content and features will be discernible without relying on color alone.

### Actual

- The slider track relies on color alone to convey its shape and size.

## Recommendations

Give the range slider's `<input>` a `1px` transparent border. This won't show up normally, but will be rendered as a solid border by the High Contrast themes.

If this has any wonky affects on layout, the [`-ms-high contrast` media query](https://msdn.microsoft.com/en-us/library/hh771830(v=vs.85).aspx) can be used, too.

### Range slider output recommendation (CSS)

```
.Polaris-RangeSlider__Input {
  margin: 0;
  padding: 0;
  width: 100%;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 20;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  border: 1px solid transparent;
}
```
Here's what that looks like:

![Range slider in High Contrast with a solid border around the track](https://user-images.githubusercontent.com/1462085/41673668-684beec8-7472-11e8-9907-46e5518a952d.png)

## Specifications

- **Component:** Range slider
- **WCAG SCs:** [1.4.1 Use of Color (Level A)](https://www.w3.org/TR/WCAG21/#use-of-color)
- **Testing done with:**
- Windows High Contrast/Edge/Windows 10

**Comment by ruairiphackett** [comment]
@dpersing I noticed I had a different result than you with the `transparent` border, it didn't show up for me (also strangely the input progress colours are showing now):

![https://screenshot.click/2019-05-14_12-53-14.png](https://screenshot.click/2019-05-14_12-53-14.png)

I'll put in a PR to add this, but it can be closed if the border is no longer necessary due to the progress colours being present:

```
.Polaris-RangeSlider__Input {
  @media (-ms-high-contrast: active) {
    border: 1px solid ms-high-contrast-color('text');
  }
]
```

![https://screenshot.click/2019-05-14_12-57-39.png](https://screenshot.click/2019-05-14_12-57-39.png)

**Comment by dpersing** [comment]
@ruairiphackett I'm seeing the track now as well. It looks like HC is parsing the color values correctly, so I'll close this out! I think they must have made an update to Edge or HC, since the values haven't changed.

**Comment by ruairiphackett** [comment]
Thanks @dpersing !
